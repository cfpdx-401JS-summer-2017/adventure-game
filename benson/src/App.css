.App {
  text-align: center;
}

.App-intro {
  font-size: large;
}

h6{
  color: black;
}

