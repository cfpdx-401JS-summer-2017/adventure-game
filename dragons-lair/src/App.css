/*
red     rgb(182, 7, 45);
orange  rgb(230, 118, 51);
yellow  rgb(244, 186, 51);
black   rgb(29, 29, 27);
*/


.main {
  min-height: 100vh;
  text-align: center;
  background-color: #222;
  padding: 20px;
  color: white;
}

button {
  font-size: 1.5rem;
  margin: .25rem;
  color: rgb(29, 29, 27);
  border: 4px solid rgb(29, 29, 27);
  border-radius: 6px;
  background: linear-gradient(to bottom, rgb(182, 7, 45) 0%, rgb(230, 118, 51) 50%, rgb(244, 186, 51) 100%);

}

button:hover {
  box-shadow: 0 5px 15px rgba(0,0,0,0.6);
}

button:focus {
  outline: none;
}

table {
  margin: 0 auto;
  text-align: center;
  border: 1px solid white;
}
