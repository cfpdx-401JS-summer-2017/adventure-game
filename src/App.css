#app p {
  margin: 1em;
}
#input {display: table; width: 100%}
#inputLeft {display: table-cell; vertical-align: top; width: 16px; padding-top: 5px;}
#inputRight {display: table-cell; vertical-align: top; width: calc(100% - 16px);}
textarea.gameInput {
  border: none;
  background: #227;
  color: #eee;
  font-family: 'Inconsolata', monospace;
  font-size: 1em;
  width: 100%;
  resize: none;
}
textarea.gameInput:focus {
  outline: none;
}
.roomMsg   { color: #ff3; }
.itemMsg   { color: #f93; }
.charMsg   { color: #f39; }
.gameMsg   { color: #6f6; }
.errorMsg  { color: #f55; }
.playerMsg { color: #bdf; }