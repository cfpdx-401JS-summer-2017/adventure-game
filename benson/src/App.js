import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import rooms from './rooms';
// import benson from './models/player.js';
import enemies from './enemies';

class App extends Component {
  constructor() {
    super();

    this.state = {
      rooms,
      room: rooms[0],
      player: {
        name: 'Benson Wigglepuff',
        inventory: ['Sailor Cap', 'Fishnet Stockings']
      }
    }
    this.handleExit = this.handleExit.bind(this);
    this.handlePickup = this.handlePickup.bind(this);
  }

  handleExit(room) {
    this.setState({room});
  }

  handlePickup(item) {
    const {room, player} = this.state;
    const index = room.items.indexOf(item);
    if(index > -1) room.items.splice(index, 1);
    player.inventory.push(item);
    this.setState({room, player})
  }

  render() {
    const {player, room} = this.state;
    return (
      <div className="App">
        <div className="App-header">
          <h2>Benson's Burger</h2>
          <h6>{player.name}</h6>
          <h6>{player.inventory.join(', ')}</h6>
        </div>
        <Room room = {room}
          onExit = {this.handleExit}
          onPickup = {this.handlePickup}
        />
      </div>
    );
  };
}

function Room({room, onExit, onPickup}) {
  return(
    <div>
      <h2>{room.name}</h2>
      <p>{room.initText}</p>
      {room.enemies &&
        <p>
          {room.enemies.enemyText}
        </p>
      }
      {room.enemies === null &&
        <p>
          {room.items.map((item, i) => (
          <button key = {i} onClick = {() => onPickup(item)}>
            Pick up {item}
          </button>
        ))}
        </p>
      }
      {room.doors.map((door, i) =>{
        return(
          <button key={i} onClick={() => onExit(door)}> 
            Door to {door.name}
          </button>
        );
      })}
    </div>
  );
}



export default App;
