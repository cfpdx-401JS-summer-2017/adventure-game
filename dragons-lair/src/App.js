import React, { Component } from 'react';
import './App.css';
import { Splash } from './game/splash.js'
import { Game } from './game/game.js'
import { GameOver } from './game/game-over.js'
import acts from './game/acts.js'

class App extends Component {

  constructor(props) {
    super(props);

    this.state = {
      gameStatus: 'splash', //'active', 'game-over'
      act: 0,
      scene: 0,
      player: {
        lives: 3,
        coins: 0,
      },
    }

    this.handleMove = this.handleMove.bind(this);
    this.goToActive = this.goToActive.bind(this);
  }

  handleMove(move) {
    console.log('user picked', move);
    let correct = acts[this.state.act].scenes[this.state.scene].correct;

    if (move === correct) {
      if(this.state.scene < (acts[this.state.act].scenes.length - 1)) {
        // go to next scene
        let scene = this.state.scene + 1;
        this.setState({ scene })
      } else {
        // go to next act
        let act = this.state.act + 1;
        this.setState({ act, scene: 0 })
      }

    } else {
      if(this.state.player.lives > 0) {
        // take away a life
        let lives = this.state.player.lives - 1;
        this.setState({ player: { lives, coins: 0 } })
      } else {
        //TODO: 
        alert('game over!');
        this.setState({ gameStatus: "game-over"});
      }
    }

  }

  goToActive() {
    this.setState({ gameStatus: 'active' });
  }

  newGame() {
    
  }

  render() {
    if(this.state.gameStatus === 'splash') return <Splash handleClick={this.goToActive} />;

    if(this.state.gameStatus === 'active')
      return (
        <Game player={this.state.player}
              act={this.state.act}
              scene={this.state.act}
              handleClick={this.handleMove} />
      )
    
    if(this.state.gameStatus === 'game-over') return <GameOver handleClick={this.goToActive} />;
  }

}

export default App;

function doTheThing() {

}