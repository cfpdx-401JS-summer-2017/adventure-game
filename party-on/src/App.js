import React, { Component } from 'react';
import './App.css';
import rooms from './rooms/Rooms';
import player from './player/Player';

class App extends Component {

  constructor(props) {
    super(props);

    this.state = {
      rooms,
      people: [],
      room: rooms[0],
      player: {}
    }

    this.handleExit = this.handleExit.bind(this);
    this.handlePickup = this.handlePickup.bind(this);
  }

  handleExit(room) {
    this.setState({room})
  }

  handlePickup(item) {
    const { room, player } = this.state;
    const index = room.items.indexOf(item);

    if( index > -1 ) room.items.splice(index, 1);
    player.items.push(item);
    this.setState(
      { room, player }
    )
  }

  render() {
    const { player, room } = this.state;

    return (

      <div className="App">
        <div className="App-header">
          <h2>Welcome to React</h2>
        </div>
        <Room room={room} onExit={this.handleExit} onPickup={this.handlePickup}/>
      </div>
    );
  }
}

function Room({ room, onExit, onPickup }){
  return (
    <div>
      <h2>{room.name}</h2>
      <p>You see: {room.items.map((item, i) => (
        <button key={i} onClick={() => onPickup(item)}>{item}</button>))}
      </p>
      {room.exits.map( (door, i) => {
        return (
          <button key={i} onClick={() => onExit(door)}>
            Door to {door.name}
          </button>)
      })}
    </div>
  )
}

export default App;
