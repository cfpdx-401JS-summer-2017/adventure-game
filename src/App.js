import React, { Component } from 'react';
import logo from './true.png';
import Room from './modules/Room';
import rooms from './modules/rooms';
import chars from './modules/chars';
import move from './modules/principal';
import {Start, Win, Lose} from './modules/screens';
import challenges from './modules/challenges';
import Challenge from './modules/Challenge';

import './App.css';

// TODO adjacent room warning

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      screen: 'startScreen',
      // win: null,
      rooms: rooms,
      playerRoom: rooms[0],
      princRoom: rooms[1],
      item: {
        key: '',
        name: ''
      },
      player: chars.player,
      principal: chars.principal,
      playerAnswer: '',
      challenge: {},
      visible: false,
      win: false
    };
    this.initialState = this.state;
    this.handlePickup = this.handlePickup.bind(this);
    this.handleRoomRelations = this.handleRoomRelations.bind(this);
    this.changeScreen = this.changeScreen.bind(this);
  }

  changeScreen() {
    if(this.state.screen === 'startScreen') this.setState({ screen: 'gameScreen' });
    else window.location.reload(true);
  }

  handleRoomRelations(playerDest) {
    this.setState({win:false});
    const { player, princRoom, rooms, playerRoom } = this.state;
    if(playerRoom !== princRoom) {
      console.log('not same room: ', 'player is in this room: ', playerRoom);
      this.setState({playerRoom: playerDest});
      let princDest = move(princRoom, playerRoom);
      this.setState({princRoom: princDest});
      // this.handleRoomRelations(playerRoom)
    }
    else if(playerRoom ===  princRoom) {
      console.log('handling: ',playerRoom, princRoom);
      console.log('testing hall pass: ', player.inventory, 'pr: ',playerRoom.key);
      if(player.inventory.includes('hall pass') && (playerRoom.key === 'westHall' || playerRoom.key === 'eastHall')) {
        console.log('hall pass = safe!');
        this.setState({princRoom: rooms[1], playerRoom: playerDest});
      } else {
        console.log('time for a challenge!');
        this.setState({playerRoom: playerDest})
         let index = Math.trunc(Math.random() * 10);
        this.setState({ visible: true });
        this.setState({ challenge: challenges[index] });
        }
    } else if (playerDest === princRoom) {
      this.setState({ playerRoom: playerDest });
    } else if (playerDest !== princRoom) {
      this.setState({ playerRoom: playerDest });
      let princDest = move(princRoom, playerDest);
      this.setState({ princRoom: princDest });
    }
    if(player.inventory.length >= 9
        && (playerDest.key === 'westHall' || playerDest.key === 'playground')) {
      console.log('Congratulations!');
      this.setState({ screen: 'winScreen' });
    }
  }

  handlePickup(item) {
    const { playerRoom, player, princRoom } = this.state;
    this.handleRoomRelations(playerRoom);
    const index = playerRoom.items.indexOf(item);
    if (index > -1) playerRoom.items.splice(index, 1);
    player.inventory.push(item);
    let princDest = move(princRoom, playerRoom);
    this.setState({ princRoom: princDest });
  }

  handleChallengeAnswer({ target }) {
    this.setState({ playerAnswer: target.value });
    const { challenge, playerAnswer, win } = this.state;
    if (target.name === 'submit') {
      if (
        playerAnswer === challenge.correctAnswer &&
        target.name === 'submit'
      ) {
        this.setState({ visible: false });
        this.setState({ win: true });
        this.setState({ princRoom: rooms[1] });
      } else {
          this.setState({ screen: 'loseScreen' });
      }
    }

  }

  render() {
    const {
      player,
      playerRoom,
      challenge,
      visible,
      playerAnswer,
      win,
      screen
    } = this.state;
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h6>{player.name}</h6>
            <h6>{player.inventory.join(', ')}</h6>
            </div>
        <div className="start">
          {this.state.screen === 'startScreen' && <Start onStart={this.changeScreen} />}
        </div>
        <div className="game">
          {this.state.screen === 'gameScreen' && visible !== true &&
          <div>
            <Room room={playerRoom}
              onExit={this.handleRoomRelations}
              onPickup={this.handlePickup}
            />
          </div>}
        {win && <div className="win">You are correct!</div>}
        {visible && screen === 'gameScreen' &&
          <Challenge
            challenge={challenge}
            value={playerAnswer}
            onSubmit={target => this.handleChallengeAnswer(target)}
          />}
          </div>
        <div className="win">
          {screen === 'winScreen' &&
          <div>
            <Win onWin={this.changeScreen} />
          </div>}
        </div>
        <div className="lose">
          {screen === 'loseScreen' &&
          <div>
            <Lose onLose={this.changeScreen} />
          </div> }
        </div>
</div>
    );
  }
}
export default App;
