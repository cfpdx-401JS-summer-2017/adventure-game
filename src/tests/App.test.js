import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import App from '../App';
import renderer from 'react-test-renderer';


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

it('renders a snapshot', () => {
  const tree = renderer.create(<App/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('player and principal start in right spots', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.state().playerRoom.key === 'westHall')
  expect(wrapper.state().princRoom.key === 'office')
})