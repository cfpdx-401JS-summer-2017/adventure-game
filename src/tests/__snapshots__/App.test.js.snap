// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="true.png"
    />
    <h6>
      Percy Prankster
    </h6>
    <h6>
      
    </h6>
  </div>
  <div
    className="start"
  >
    <div
      className="startWrapper"
      style={
        Object {
          "backgroundColor": "white",
          "height": "100%",
          "width": "100%",
          "zIndex": "100",
        }
      }
    >
      <h2>
        Welcome to the Game of School
      </h2>
      <p>
        You are the prime jokester at your school, but your glory days may be over. Unless you can gather all evidence of your pranks and dispose of it quickly, the principal will suspend you for good.
      </p>
      <p>
        Search the rooms of your school for incriminating evidence. Pick up all of the items to win the game. If the principal catches you, use your wits to get free, or else you lose!
      </p>
      <button
        onClick={[Function]}
      >
        Play the Game!
      </button>
      <br />
      <img
        alt="Map"
        src="./worldMap.png"
        style={
          Object {
            "maxWidth": 700,
          }
        }
      />
    </div>
  </div>
  <div
    className="game"
  />
  <div
    className="win"
  />
  <div
    className="lose"
  />
</div>
`;
